ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh

```diff
@@ -108,105 +110,89 @@ function disable_default_catalogsource () {
-    # Running Command: oc adm catalog mirror -a "/tmp/new-dockerconfigjson" ec2-3-90-59-26.compute-1.amazonaws.com:5000/openshift-qe-optional-operators/aosqe-index:v4.12 ec2-3-90-59-26.compute-1.amazonaws.com:5000 --continue-on-error --to-manifests=/tmp/olm_mirror
-    # error: unable to read image ec2-3-90-59-26.compute-1.amazonaws.com:5000/openshift-qe-optional-operators/aosqe-index:v4.12: Get "https://ec2-3-90-59-26.compute-1.amazonaws.com:5000/v2/": x509: certificate signed by unknown authority
-    run_command "oc adm catalog mirror --manifests-only --insecure=true --continue-on-error=true --skip-verification=true -a \"/tmp/new-dockerconfigjson\" ${origin_index_image} ${MIRROR_REGISTRY_HOST} --to-manifests=/tmp/olm_mirror" || ret=$?
-    if [[ $ret -eq 0 ]]; then
-        echo "get manifests successfully"
-        # There are some images with the registry.redhat.io prefix, but they are not store there in fact, such as unreleased images, as follows,
-        # error: unable to retrieve source image registry.redhat.io/openshift-logging/cluster-logging-rhel9-operator manifest sha256:2801ca51e913273d7db6728f05cb50f18b5c0062f95e92609645b592512aed21: manifest unknown: manifest unknown
-        # so, replace registry.redhat.io with brew.registry.redhat.io.
-        run_command "sed -i 's/^registry.redhat.io/brew.registry.redhat.io/g' /tmp/olm_mirror/mapping.txt"
-        run_command "oc image mirror --insecure=true --skip-missing=true --continue-on-error=true --skip-verification=true -a \"/tmp/new-dockerconfigjson\" -f /tmp/olm_mirror/mapping.txt" || ret=$?
-        if [[ $ret -eq 0 ]]; then
-            echo "mirror optional operator images succeed!"
-        else
-            run_command "ls -l /tmp/olm_mirror/"
-            run_command "cat /tmp/olm_mirror/imageContentSourcePolicy.yaml"
-            run_command "cat /tmp/olm_mirror/mapping.txt"
-        fi
-    else
-        echo "fail to get optional operators' manifests!"
-        return 1
+    work_dir="/tmp"
+    #https://docs.openshift.com/container-platform/4.16/installing/disconnected_install/installing-mirroring-disconnected.html
+    #The default XDG_RUNTIME_DIR /var/run can not be edit in a pod. Replace /var/run with /tmp in this script
+    export XDG_RUNTIME_DIR=$work_dir
+    export REGISTRY_AUTH_FILE="$XDG_RUNTIME_DIR/containers/auth.json"
+    # skopeo login create/update the REGISTRY_AUTH_FILE
+    echo "REGISTRY_AUTH_FILE is $REGISTRY_AUTH_FILE"
+    skopeo login ${MIRROR_REGISTRY_HOST} -u ${mirror_registry_user} -p ${mirror_registry_password} --tls-verify=false
+    skopeo login brew.registry.redhat.io -u ${brew_auth_user} -p ${brew_auth_password} --tls-verify=false
+    skopeo login registry.redhat.io -u ${redhat_auth_user} -p ${redhat_auth_password}
+    skopeo login quay.io/openshift-qe-optional-operators -u ${optional_auth_user} -p ${optional_auth_password}
+
+    echo "skopeo copy docker://${origin_index_image} oci:///${work_dir}oci-local-catalog --remove-signatures"
```

Comment:
here I think should be : `oci://${work_dir}/oci-local-catalog` ,  if we support arm , also need `--all` for skopeo command :  `skopeo copy --all`